alchemist-composeui/src/jvmMain/kotlin/it/unibo/alchemist/boundary/composeui/ComposeMonitor.kt

```diff
@@ -10,21 +10,33 @@
 package it.unibo.alchemist.boundary.composeui
 
 import androidx.compose.ui.window.Window
-import androidx.compose.ui.window.application
+import androidx.compose.ui.window.awaitApplication
 import it.unibo.alchemist.boundary.OutputMonitor
 import it.unibo.alchemist.model.Environment
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlin.concurrent.thread
 
 /**
  * Monitor extension that uses JVM Compose UI to display the simulation.
  */
 class ComposeMonitor : OutputMonitor<Any, Nothing> {
     override fun initialized(environment: Environment<Any, Nothing>) {
-        application {
-            Window(
-                onCloseRequest = { },
-                title = "Alchemist",
-            ) {
-                app()
+        thread(
+            name = "ComposeMonitor",
+            isDaemon = true,
+        ) {
```

Comment:
Instead of launching a thread, have you considered using a different dispatcher in the `launch`?
Dealing with dispatchers is the usual way to go if you do not have strong reasons to spawn platform threads.

Reply:
It is not clear to me the attempt you performed. Could you paste here an MRE?
AFAIK spawning a coroutine does not fork the main process, instead, it schedules the coroutine into a different thread pool (if used a dispatcher)

Reply:
I was aware of this too. However it seems that issue resides on how the Monitors get spawned.

```kotlin
class ComposeMonitor : OutputMonitor<Any, Nothing> {
    override fun initialized(environment: Environment<Any, Nothing>) {
        application {
            Window(onCloseRequest = ::exitApplication, title = "Alchemist") { App() }
        }
    }
}
```

Launching this snippet, the simulation freezes, haven't been able to figure out why.